src/import.c

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "storage.h"
+#include "util.h"
+#include "module.h"
+#include "import.h"
+#include "entry.h"
+#include <string.h>
+#include <stdio.h>
+
+
+/**
+ * CacheImport:
+ * @shadow_path: Path to shadow file.
+ * @shadow_hash: Hashed password. 
+ * @storage_path: Path to store policy cache entry.
+ * @username: Username to create cache entry for.
+ * @error: Errors.
+ *
+ * Returns: #True if CacheEntry was created or #False on failure.
+ */
+
+gboolean CacheImport(const char *shadow_path, const char *storage_path, const char *username, GError **error) {
+  if (!shadow_path) {
+    shadow_path = "/etc/shadow";
+  }
+  
+  if (!storage_path) {
+    storage_path = DEFAULT_STORAGE_PATH;
+  }
+
+  GError *errors = NULL;
+  // Get user's hashed password.
+  GBytes* shadow_hash = ReadShadowFile(shadow_path, username, &errors);
+
+  // Create a new Cache Entry.
+  CacheEntry *entry = CacheEntryNew();
+
+  // Set the newly created entry hash using the /etc/shadow hash.
+  CacheEntryHashSet(entry, CACHE_ENTRY_ALGORITHM_CRYPT, shadow_hash, NULL);
+  GError *tmp_error = NULL;
+  CacheStorage *storage = CacheStorageNew(storage_path);
+  if (!CacheStoragePutEntry(storage, username, entry, &tmp_error))
+      goto done;
+
+  done:
+    if (tmp_error) {
```

Comment:
To make sure I'm understanding you.  I removed the if statement and nested the g_set_error under the CacheStoragePutEntry statement.

  if (!CacheStoragePutEntry(storage, username, entry, error)) {
    g_set_error(error, CACHE_IMPORT_ERROR, CACHE_IMPORT_ERROR_STORAGE_FAIL, "Could not store cache entry.");
    return FALSE;
  }
  return TRUE;
}